### ISSUE TITLE
Understanding the Difference between Normal Training vs eager mode training in TFOD API v2

### ISSUE THREAD

**Issue by Deep-learning-practitioner** [tensorflow/tensorflow#43097]
Eager mode allowed you to see tensor values when TF 1 used sessions and graphs but since eager mode is the default in tf 2.0 so its a little confusing.

I want to know the differences between the normal training vs Eager mode training.  One clear difference I can see is that the code is a lot different for both of them. Even the final data format is not in tfrecords in eager mode.  And the code is bigger.

**I have the following questions:**

- So is the Eager Mode Method Faster (Since it can train a really basic detector in just 5 minutes and the normal method takes just a few minutes to initialize)?

- If the params of pipeline.config are not changed then will the final model have the same accuracy regardless of which method was used (normal or eager) and again will the training time be different? 

- What's the advantage of eager mode vs normal training, is it more flexibility in training?

- Going forward what method will you recommend new users to use? and when will you prefer one method over the other?

_Thanks, would really appreciate if these questions can be answered as I can't find any docs that address these questions._

**Comment by bhack** [comment]
If you are coming from TF 1.x I suggest you to give an overview to:
https://medium.com/ai%C2%B3-theory-practice-business/tensorflow-1-0-vs-2-0-part-1-computational-graphs-4bb6e31c1a0f
https://medium.com/ai%C2%B3-theory-practice-business/tensorflow-1-0-vs-2-0-part-2-eager-execution-and-autograph-47473ed8b817

**Comment by Deep-learning-practitioner** [comment]
> 
> 
> If you are coming from TF 1.x I suggest you to give an overview to:
> https://medium.com/ai%C2%B3-theory-practice-business/tensorflow-1-0-vs-2-0-part-1-computational-graphs-4bb6e31c1a0f
> https://medium.com/ai%C2%B3-theory-practice-business/tensorflow-1-0-vs-2-0-part-2-eager-execution-and-autograph-47473ed8b817

@bhack I read the articles above and also 2 other parts in the series. Now I do think that eager mode in TFOD API is more useful when you want to see the whole training process in detail and want to make debugging easier also in this mode no computational graphs will be used but I still have 2 questions above unanswered:


1.  How is the eager mode training faster (Since it can train a really basic detector in just 5 minutes and the normal method takes just a few minutes to initialize)?  is it because of the initializations done for the computational graph?

2.  If the params of pipeline.config are not changed then will the final model have the same accuracy and latency regardless of which method was used (normal or eager) and again will the training time be different?

If you're unable to answer these can you please mention someone in the tfod Dev team.

**Comment by Deep-learning-practitioner** [comment]
@bhack Hi could really use some insight on the above 2 questions

**Comment by bhack** [comment]
I don't know your specific model but eager is not generally faster:
https://www.tensorflow.org/guide/intro_to_graphs#the_benefits_of_graphs

Please use [Stackoverflow](https://stackoverflow.com/questions/tagged/tensorflow) for this type of questions.

**Comment by Deep-learning-practitioner** [comment]
> 
> 
> I don't know your specific model but eager is not generally faster:
> https://www.tensorflow.org/guide/intro_to_graphs#the_benefits_of_graphs
> 
> Please use [Stackoverflow](https://stackoverflow.com/questions/tagged/tensorflow) for this type of questions.

Thanks for replying, yeah but when the question is too specific It's really hard to get answers from there.

**Comment by sachinprasadhs** [comment]
Usually Eager mode is slower compared to graph mode, to make the best use of performance and optimization you can use Autograph in Tensorflow 2.x , once you implement part of your code with Autograph using @tf.function decorator, that part of code will be executed in graph mode. See more details [here](https://github.com/keras-team/keras/issues/15545).
